roxctl/common/flags/endpoint_test.go

```diff
@@ -0,0 +1,85 @@
+package flags
+
+import (
+	"testing"
+
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEndpointAndPlaintextSetting(t *testing.T) {
+	truth := false
+	endpointChanged = &truth
+	plaintextSet = &truth
+
+	testCases := []struct {
+		endpoint     string
+		host         string
+		usePlaintext bool
+		err          string
+	}{
+		{
+			endpoint: "localhost:8443",
+			host:     "localhost:8443",
+		},
+		{
+			endpoint: "https://localhost",
+			host:     "localhost:443",
+		},
+		{
+			endpoint: "https://example.com:443",
+			host:     "example.com:443",
+		},
+		{
+			endpoint:     "http://example.com:80",
+			host:         "example.com:80",
+			usePlaintext: true,
+		},
+		{
+			endpoint: "https://example.com",
+			host:     "example.com:443",
+		},
+		{
+			endpoint:     "http://example.com",
+			host:         "example.com:80",
+			usePlaintext: true,
+		},
+		{
+			endpoint:     "http://128.66.0.1",
+			host:         "128.66.0.1:80",
+			usePlaintext: true,
+		},
+		{
+			endpoint: "128.66.0.1",
+			err:      "invalid endpoint: address 128.66.0.1: missing port in address, the scheme should be: http(s)://<endpoint>:<port>",
```

Comment:
I'm not a fan of checking for exact error message matches though we do this in our code. I'd much rather check the error class (invalid args). I sort of see why you do this so maybe let's do both:
* check the error class
* check that "missing port" substring exists in the error message.

But maybe leaving this as-is is simplest.